app/lib/package/overrides.dart

```diff
@@ -19,6 +19,17 @@ const redirectDartdocPages = <String, String>{
   'flutter': 'https://docs.flutter.io/',
 };
 
+/// Known packages that should be put in `dev_dependencies`
+const devDependencyPackages = <String>{
```

Comment:
Can we leave a comment here stating that this is a temporary hack that will be removed when:
https://github.com/dart-lang/pub-dev/issues/3403 is probably implemented?